tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -453,7 +464,7 @@ public:
          std::string msg = "Cannot jit Snapshot call. Interpreter error code is " + std::to_string(errorCode) + ".";
          throw std::runtime_error(msg);
       }
-      return *reinterpret_cast<TInterface<TLoopManager> *>(newTDFPtr);
+      return *reinterpret_cast<TResultPtr<TInterface<TLoopManager>> *>(newTDFPtr);
```

Comment:
this is a problem that was already present, but isn't the lifetime of this `TResultPtr` the same as ROOT's lifetime? it's basically hogging a tiny bit of memory

Reply:
Comment noted. We can get rid of this widespread tiny memory hoarding later.